### PR TITLE
removed redundant tests

### PR DESCRIPTION
**Pull request by nzediegwu1**
*rewrote tests to handle new validations in words function

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+﻿const mocha = require('mocha');
+const chai = require('chai');
+const program = require('../words');
+const assert = chai.assert;
+chai.config.truncateThreshold = 0;
```
Comment (oreoluwade) [inline test/test.js]: What does this line of code do?

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+describe('Test words function', () => {
+    it('should handle specified input', () => {
```
Comment (oreoluwade) [inline test/test.js]: The wording of the describe block could be more descriptive. Something like: "It should return the words of an input sentence and their frequencies as key and value respectively of a JSON object."

It doesn't have to be as long and complicated as what I wrote, but at least, it should be descriptive enough for someone who probably doesn't know much about your function/code in the first place. The same applies to the other` describe` and `it` blocks.

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        let testcase = program.words('olly olly in come free');
+        assert.deepEqual(testcase, { olly: 2, in: 1, come: 1, free: 1 });
+    });
+    it('should handle more complex input', () => {
```
Comment (oreoluwade) [inline test/test.js]: This test case is not exactly different from the first test case. It isn't more complex. It only has more words. If you wanted to specify a more complex sentence, perhaps, you'd have configured your function to handle `special characters` or `numbers` sandwiched in between valid words in the sentence, then you either make the function ignore them or recognize them as well. Reply on this thread if you need more clarification about this.

File: test/test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        let testcase = program.words('olly olly has come in the free has the free olly');
+        assert.deepEqual(testcase, { 'olly': 3, 'has': 2, 'come': 1, 'in': 1, 'the': 2, 'free': 2 });
+    });
+    it('should return an object', () => {
+        let testCase = typeof program.words('olly olly has come in the free has the free olly');
+        assert.deepEqual(testCase, 'object');
+    });
+    it('should handle multiple spaces', () => {
```
Comment (oreoluwade) [inline test/test.js]: The `it` block here could better be phrased as "It should strip off extra unwanted spaces in any input sentence/parameter"

File: test/test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        let testcase = program.words('olly  olly car    car');
+        assert.deepEqual(testcase, { 'olly': 2, 'car': 2 });
+    });
+    it('test that empty/space string return empty object', () => {
```
Comment (oreoluwade) [inline test/test.js]: The `it` block read out with the string you provided as description should flow as a normal English sentence. Picture this: `it test that empty/space string return empty object` and compare with: `it should return an empty object when an empty string is supplied as the input parameter/sentence`. What do you think?

Apply that to other test cases as well.

File: test/test.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    });
+    it('test that parameter is not undefined', () => {
+        let testcase = program.words();
+        assert.deepEqual(testcase, { error: 'parameter is undefined' });
+    });
+    it('test that parameter must not be an object', () => {
+        let testcase = program.words({a:'apple', b:'boy'});
+        assert.deepEqual(testcase, { error: 'parameter must not be an object' });
+    });
+    it('should handle uppercase and lower case as same word', () => {
```
Comment (oreoluwade) [inline test/test.js]: You can see that the wording here is fine as it is. So please, go ahead and make the others better as well. :)

**Review by oreoluwade — CHANGES_REQUESTED** [review]
I can still see that there is no package.json present on this branch as well. So there is no way for me to install dependencies or even go ahead to run your tests. Please, fix that.

**Comment by nzediegwu1** [comment]
Package.json has been added. Thank you

File: test/test.js (hunk 1 above)
Comment (nzediegwu1) [reply test/test.js]: It was a code it was trying to use to fix the issue the problem of assert.equal on arrays and objects in my previous implementation before I eventually resorted to Json.stringify. would remove asap

File: test/test.js (hunk 2 above)
Comment (nzediegwu1) [reply test/test.js]: Thanks a lot. Will make necessary changes asap

File: test/test.js (hunk 5 above)
Comment (nzediegwu1) [reply test/test.js]: Much better and eloquent, would makes changes immediately

File: test/test.js (hunk 3 above)
Comment (nzediegwu1) [reply test/test.js]: Well understood. Thanks for the hint. I thought some words might complicate issues like in the sentence: "Friend's house friend", hence I decided to ignore special characters. Would therefore prefer to remove this test case since it seems redundant.

File: test/test.js (hunk 4 above)
Comment (nzediegwu1) [reply test/test.js]: Thanks a lot. Would rephrase immediately
